File: src/client_service.cpp
#include "ros/ros.h"
#include "service_example/Archivo_var.h"
#include <cstdlib>

int main(int argc, char **argv) {

  ros::init(argc, argv, "service_server");
  ros::NodeHandle n;
  ros::ServiceClient client = n.serviceClient<service_example::Archivo_var>("filtro");
  service_example::Archivo_var srv;

  std::vector<float> a(2);

  a.push_back(1.0);
  a.push_back(2.0);

  srv.request.valores = a;
  if (client.call(srv))
  {
    ROS_INFO("Resultado del filtrado: %f", (float)srv.response.resultado);
  } 
  else {
    ROS_ERROR("Fallo el llamado al servicio del filtrado");
    return 1;
  }
  return 0;
}

File: src/client_service.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Float64MultiArray
from service_example.srv import Archivo_var

valor_anterior = 0

if __name__ == "__main__":

  rospy.init_node('service_client')
  rospy.wait_for_service('filtro')
  client = rospy.ServiceProxy('filtro', Archivo_var)

  array = [23.2,22.1]
  variable = client(array)
  resultado = variable.resultado
  print("Resultado del filtro: "+str(resultado))

# Archivo_var.srv
# float32[] valores
# ---
# float32 resultado
